lib/nerves_hub_link/update_manager.ex

```diff
@@ -186,43 +191,42 @@ defmodule NervesHubLink.UpdateManager do
     # interrupt FWUP and let the task finish. After update and reboot, the
     # device will check-in and get an update message if it was actually new and
     # required
-    state
+    {:ignored, state}
   end
 
   defp maybe_update_firmware(%UpdateInfo{} = update_info, fwup_public_keys, %State{} = state) do
-    # Cancel an existing timer if it exists.
-    # This prevents rescheduled updates`
-    # from compounding.
-    state = maybe_cancel_timer(state)
-
     # possibly offload update decision to an external module.
     # This will allow application developers
     # to control exactly when an update is applied.
     # note: update_available is a behaviour function
     case state.fwup_config.update_available.(update_info) do
       :apply ->
-        start_fwup_stream(update_info, fwup_public_keys, state)
+        NervesHubLink.send_firmware_update_status(:updating)
+        {{:updating, 0}, start_fwup_stream(update_info, fwup_public_keys, state)}
 
       :ignore ->
-        state
+        NervesHubLink.send_firmware_update_status(:ignored)
+        Logger.info("[NervesHubLink] ignoring firmware update request")
+        {:ignored, %{state | status: :idle, previous_status: :ignored}}
 
-      {:reschedule, ms} ->
-        timer =
-          Process.send_after(self(), {:update_reschedule, update_info, fwup_public_keys}, ms)
-
-        Logger.info("[NervesHubLink] rescheduling firmware update in #{ms} milliseconds")
-        %{state | status: :update_rescheduled, update_reschedule_timer: timer}
-    end
-  end
+      {:reschedule, minutes, :minutes} ->
+        until =
+          DateTime.utc_now()
+          |> DateTime.add(minutes, :minute)
 
-  defp maybe_update_firmware(_, _, state), do: state
+        NervesHubLink.send_firmware_update_status(:reschedule, %{until: until})
+        Logger.info("[NervesHubLink] rescheduling firmware update in #{minutes} minutes")
+        {:rescheduled, %{state | status: :idle, previous_status: :rescheduled}}
 
-  defp maybe_cancel_timer(%{update_reschedule_timer: nil} = state), do: state
-
-  defp maybe_cancel_timer(%{update_reschedule_timer: timer} = state) do
-    _ = Process.cancel_timer(timer)
+      {:reschedule, ms} ->
+        until =
+          DateTime.utc_now()
+          |> DateTime.add(round(ms / 1_000), :second)
```

Comment:
```suggestion
          |> DateTime.add(ms, :millisecond)
```
I don't think you need to round. Just add the expected time here

Reply:
True true. For some reason I didn't click that I could use `:millisecond`